src/geo/map/legends/custom-legend-model.js

```diff
@@ -7,6 +7,24 @@ var CustomLegendModel = StaticLegendModelBase.extend({
       type: 'custom',
       items: []
     });
+  },
+
+  parse: function (r, opts) {
```

Comment:
Not sure if I understand completely. Do you mean something like this?

```
initialize: function (attrs, opts) {
    var attributes = _.extend({},
      _.omit(attrs, 'items')
    );
    attributes.items = [];

    if (attrs.items) {
      attributes.items = attrs.items.map(function (item) {
        return {
          name: item.title,
          color: item.color
        };
      });
    }

    this.set(attributes);
  }
```

If so, I don't like it because we are setting the attributes twice due to Backbone already set attributes before to call `initialize`. And the source code itself defines:
```
if (options.parse) attrs = this.parse(attrs, options) || {};
```
in the Model constructor in order to deal with this scenario. `parse` is defined as a no-op, so if a model doesn't implement it, there is no harm.